src/object/endpoint.c

```diff
@@ -226,10 +226,10 @@ void receiveIPC(tcb_t *thread, cap_t cap, bool_t isBlocking)
             do_call = thread_state_ptr_get_blockingIPCIsCall(&sender->tcbState);
 
 #ifdef CONFIG_KERNEL_MCS
-            if (do_call ||
-                seL4_Fault_get_seL4_FaultType(sender->tcbFault) != seL4_Fault_NullFault) {
+            word_t faultType = seL4_Fault_get_seL4_FaultType(sender->tcbFault);
```

Comment:
the original code here optimised by not reading the fault type at all if `do_call` was set. Please put this back.

Reply:
Is this also why `do_call` is not set by the call to `sendIPC` in `sendFaultIPC`, i.e. so `do_call` implies that the IPC is not the result of a fault?

Reply:
do_call implies that we are doing a call like operation which generates a reply. So Call or a fault.

Reply:
In MCS, `do_call` is set to false in `sendFaultIPC` but not on master, so it does actually seem to imply that there isn't a fault.